File: hmvvm/src/main/java/com/see99/hmvvm/dagger/component/AppComponent.java
package com.see99.hmvvm.dagger.component;

import android.app.Application;

import com.see99.hmvvm.MyApp;
import com.see99.hmvvm.dagger.module.AppModule;
import com.see99.hmvvm.model.http.HttpManager;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;

@Singleton
@Component(modules = {AppModule.class})
public interface AppComponent {

    void inject(MyApp mApp);

    Application getMyApp();

    HttpManager getHttpManager();

    @Component.Builder
    interface Builder{

       @BindsInstance
       Builder application(Application application);

       AppComponent build();
    }
}

File: hmvvmkotlin/src/main/java/com/see99/hmvvmkotlin/dagger/component/AppComponent.kt
package com.see99.hmvvmkotlin.dagger.component

import android.app.Application
import com.see99.hmvvmkotlin.MyApp
import com.see99.hmvvmkotlin.dagger.module.AppModule
import com.see99.hmvvmkotlin.model.http.ApiService
import dagger.BindsInstance
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [AppModule::class])
interface  AppComponent {

    fun inject(myApp: MyApp)

    fun application():Application

    fun appServiec():ApiService


    @Component.Builder
    interface Builder{

        @BindsInstance
       fun application(application:Application) : Builder

//        fun appModule(model : AppModule) :Builder

        fun build() : AppComponent
    }
}